src/record.rs

```diff
@@ -80,6 +80,52 @@ impl Meta {
     }
 }
 
+pub(crate) struct PartiallySerializedRecord {
+    buf: Vec<u8>,
+    offset_pos: usize,
+    header_checksum_pos: usize,
+    header_start: usize,
+    header_len: usize,
+}
+
+impl PartiallySerializedRecord {
+    fn new(
+        buf: Vec<u8>,
+        offset_pos: usize,
+        header_checksum_pos: usize,
+        header_start: usize,
+        header_len: usize,
+    ) -> Self {
+        Self {
+            buf,
+            offset_pos,
+            header_checksum_pos,
+            header_start,
+            header_len,
+        }
+    }
+
+    pub(crate) fn into_serialized_with_header_checksum(
```

Comment:
It is super critical to have a test for this function. It should compare serialization results between old and new aproaches

Reply:
There is a test for header now, which is the only new logic

Reply:
I think an additional test required, that will compare raw bytes after serialization with old and with new approach

Reply:
So we keep old approach just for the tests?

Reply:
There is no need to keep the old approach. You can reproduce the old serialization logic right inside the test function. This will be a backward compatibility test. You need to create a record and make its clone. Serialize one copy using the old approach: set the offset, update the checksum, write the header, metadata and data to the buffer. Then serialize the second copy by calling new functions. Finally, compare the resulting arrays